File: coupon-common/src/main/java/com/kakao/mobility/kmcoupon/domain/member/SecurityUser.java
package com.kakao.mobility.kmcoupon.domain.member;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;

public class SecurityUser implements UserDetails {
    private Long memberId;

    private String username;
    private String password;
    private Collection<? extends GrantedAuthority> authorities;

    public SecurityUser() {
    }

    public SecurityUser(Long memberId, String username, String password, Collection<? extends GrantedAuthority> authorities) {
        this.memberId = memberId;
        this.username = username;
        this.password = password;
        this.authorities = authorities;
    }

    public Long getMemberId() {
        return memberId;
    }

    @Override
    public String getUsername() {
        return username;
    }

    @Override
    public String getPassword() {
        return password;
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return authorities;
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }
}

File: coupon-common/src/main/java/com/kakao/mobility/kmcoupon/domain/member/SecurityUser.kt
package com.kakao.mobility.kmcoupon.domain.member

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.UserDetails

// java 변환시 1개 constructor 생성
// kotlin-jpa 플러그인에 의해 entity 관련 클래스에 대해서 default 제공
class SecurityUser(
    val memberId: Long,
    private val username: String,
    private val password: String,
    private val authorities: Collection<GrantedAuthority> // 다른 방식으로
) : UserDetails {
    // never null 이기에 확정형으로
    override fun getUsername(): String {
        return username
    }

    override fun getPassword(): String {
        return password
    }

    override fun getAuthorities(): Collection<GrantedAuthority> {
        return authorities
    }

    override fun isAccountNonExpired(): Boolean {
        return true
    }

    override fun isAccountNonLocked(): Boolean {
        return true
    }

    override fun isCredentialsNonExpired(): Boolean {
        return true
    }

    override fun isEnabled(): Boolean {
        return true
    }
}